### PR TITLE
TST: Improve requires grad testing:

### PR DESCRIPTION
**Pull request by BenjaminBossan**
## Description

Previously, the corresponding tests were testing only whether specific parameters had `requires_grad` `True` or `False`. Now, all parameters are being checked. This is more rigorous.

Also, tests for `Embedding`, `Conv1D`, `Conv2d` were added (LoRA and IA³), thus superseding PR #1115.

Finally, tests for LoHa and LoKr were added.

## Note

I considered moving the tests to a separate module, as they were getting quite big and this would help with readability. For now, I left them in the same module because it leads to a better diff view and is thus easier to review. LMK if I should move the tests to a separate file.

### CODE REVIEW

**Review by pacman100 — APPROVED** [review]
Thank you @BenjaminBossan for the extensive tests to check that gradients are being set properly!

File: tests/test_custom_models.py [hunk 1]
```diff
@@ -941,45 +941,64 @@ class RequiresGradTester(unittest.TestCase):
 
     """
 
+    def check_requires_grad(self, model, *params_expected: str):
```
Comment (pacman100) [inline tests/test_custom_models.py:944]: Neat!
